index.js

```diff
@@ -51,7 +52,12 @@ class Conf {
 				const pkgPath = pkgUp.sync(parentDir);
 				// Can't use `require` because of Webpack being annoying:
 				// https://github.com/webpack/webpack/issues/196
-				options.projectName = pkgPath && JSON.parse(fs.readFileSync(pkgPath, 'utf8')).name;
+				const pkg = pkgPath && JSON.parse(fs.readFileSync(pkgPath, 'utf8'));
+
+				if (pkg) {
+					options.projectName = pkg.name;
+					options.projectVersion = pkg.version;
+				}
 			}
 
 			if (!options.projectName) {
```

Comment:
I think we have an issue here: If I specify the `projectName` option manually, the `options.projectVersion` default value assignment based on the package.json version would never be performed.

This means that the code on line `59` (`options.projectVersion = pkg.version;`) needs to be run independently of having `options.projectName` set manually or not.

Reply:
You're right. I abstracted the `pkg` logic into a legible function and re-used it in 2 separate code blocks. 


I wanted to also add a test to cover this situation you described so it doesn't happen in the future. However, it seems that there are no actual tests for the inferring values from the package.json.

Maybe this is something I can update when we add a stubbing lib?

Reply:
Just something to have in consideration is that this will read the package.json 2x in the worst case scenario, which is when there’s no name nor version defined. The worst case scenario is the most common one, so perhaps we should optimize for it by reading the package data if either the name or version is missing, assign it to a const, and then use it as we are already doing.

Reply:
Another remark: perhaps you can remove the if conditions when assigning options.projectName and options.projectVersion, and just assign them directly even if they are falsy.

Reply:
Yes, you're right. Done :)

I was afraid of committing the same error as in https://github.com/sindresorhus/conf/pull/49#discussion_r279627139, but it actually doesn't impact anything regarding this.

Thanks for catching that up.

Reply:
Actually, the problem still exists if there `options.cwd` was specified. We need to move it all way up to when the options and their defaults are being set.

Moreover,, you might want to avoid reading the package.json if not necessary (when cwd or projectName were specified AND projectVersion was specified as well). This is minor, but might be worth doing if the code remains simple.